tools/rustfmt/srcs/lib.rs

```diff
@@ -5,16 +5,20 @@ use std::path::{Path, PathBuf};
 /// The expected extension of rustfmt manifest files generated by `rustfmt_aspect`.
 pub const RUSTFMT_MANIFEST_EXTENSION: &str = "rustfmt";
 
-/// Generate an absolute path to a file without resolving symlinks
-fn absolutify_existing<T: AsRef<Path>>(path: &T) -> std::io::Result<PathBuf> {
-    let absolute_path = if path.as_ref().is_absolute() {
-        path.as_ref().to_owned()
-    } else {
-        std::env::current_dir()
-            .expect("Failed to get working directory")
-            .join(path)
-    };
-    std::fs::metadata(&absolute_path).map(|_| absolute_path)
+pub fn current_dir_name() -> PathBuf {
+    PathBuf::from(std::env::current_dir().unwrap().file_name().unwrap())
```

Comment:
General runfiles question:

Runfiles seem to be passed in/specified as e.g. `external/rust_windows_x86_64/bin/rustfmt.exe`, but sometimes (concretely, when using manifest-based runfiles) they're expected to be resolved with a workspace name prefix (`rules_rust/external/rust_windows_x86_64/bin/rustfmt.exe`, in this example, with a `rules_rust` prefix)

Where is that prefix supposed to come from? Is client code/the runfiles library supposed to assume the workspace name, or divine it somehow? This happens to coincide with the basename of the CWD for both runfiles implementations - hence this sketchy function here - but I'd like to better understand how runfiles are supposed to work r.e. this workspace prefix.

Reply:
This is a good question. I don't think the prefix is expected to be passed anywhere. Bazel uses a [launcher](https://github.com/bazelbuild/bazel/blob/23f171e2208700dc2f2f65f5ccb32bacd7aec87d/src/tools/launcher/launcher.cc#L239-L249) to set some environment variables dictating the locations of runfiles related paths. My experience is that bazel suggests paths to be treated as relative paths from the runfiles directory which is also suggested to be a relative path from the binary. I say "suggest" here since Bazel produces outputs this way but it's obviously up to the author to maintain this if the outputs are distributed in some way.

... This very likely doesn't answer your question. Maybe rephrase and I can try to be more direct 😅

Reply:
cc @meteorcloudy